heir_py/mlir_emitter.py

```diff
@@ -191,6 +312,49 @@ def emit_branch(self, branch):
     condvar = self.get_name(branch.cond)
     return f"cf.cond_br {condvar}, ^bb{branch.truebr}, ^bb{branch.falsebr}"
 
+  def emit_loop(self, target):
+    # Note right now loops that use the %i value directly will need an index_cast to an i64 element.
+    loop = self.loops[target]
+    resultvar = self.get_or_create_name(target)
+    itvar = self.get_or_create_name(loop.header.phi_var)  # create var for i
+    for_str = (
+        f"affine.for {itvar} = {loop.range.start} to {loop.range.stop} step"
+        f" {loop.range.step}"
+    )
+
+    if len(loop.inits) == 1:
+      # Note: we must generalize to inits > 1
+      init_val = self.get_name(loop.inits[0])
+      # Within the loop, forward the name for the init val to a new temp var
+      iter_arg = "itarg"
```

Comment:
This name will need to be uniquified, no? Maybe give it a name that includes the block id.

Reply:
Ah yes, right now since there's no nested loop support it doesn't since its scoped to that region, but you're right that I need to change this to future-proof the code.